pxr/imaging/hdSt/resourceRegistry.cpp

```diff
@@ -122,6 +128,10 @@ HdStResourceRegistry::HdStResourceRegistry(Hgi * const hgi)
     // default aggregation strategy for single buffers (for nested instancer)
     , _singleAggregationStrategy(
         std::make_unique<HdStVBOSimpleMemoryManager>(this))
+#ifdef PXR_MATERIALX_SUPPORT_ENABLED
+    , _materialXShaderRegistry(
+        std::make_unique<HdSt_MaterialXShaderRegistry>(mtlxCacheDirPath))
```

Comment:
The MaterialX shader registry is now owned by unique pointer to simplify include dependencies.